polkadot-parachains/rococo-runtime/src/lib.rs

```diff
@@ -595,10 +592,22 @@ struct CheckInherents;
 
 impl cumulus_pallet_parachain_system::CheckInherents<Block> for CheckInherents {
 	fn check_inherents(
-		_: &[UncheckedExtrinsic],
-		_: &cumulus_pallet_parachain_system::RelayChainStateProof,
+		block: &Block,
+		relay_state_proof: &cumulus_pallet_parachain_system::RelayChainStateProof,
 	) -> sp_inherents::CheckInherentsResult {
-		sp_inherents::CheckInherentsResult::new()
+		let relay_chain_slot = relay_state_proof
+			.read_slot()
+			.expect("Could not read the relay chain slot from the proof");
```

Comment:
This check depends on the relay state proof which came in through the parachain inherent.

Maybe a dumb question, but do the validators check that the data in the parachain inherent reflects the real relay chain state?

Reply:
Yes